numba/tests/test_random.py

```diff
@@ -1374,6 +1380,92 @@ def test_dirichlet_exceptions(self):
                 str(raises.exception),
             )
 
+class TestRandomNoncentralChiSquare(BaseTest):
+
+    def _check_sample(self, size, sample):
+
+        """Check output structure"""
+        if size is not None:
+            self.assertIsInstance(sample, np.ndarray)
+            self.assertEqual(sample.dtype, np.float64)
+            
+            if type(size) is int:
+                self.assertEqual(sample.shape, (size,))
+            else:
+                self.assertEqual(sample.shape, size)
+        else:
+             self.assertIsInstance(sample, float)
+
+        """Check statistical properties"""
+        for val in np.nditer(sample):
+            self.assertGreaterEqual(val, 0)
+
+    def test_noncentral_chisquare_default(self):
+        """
+        Test noncentral_chisquare(df, nonc, size=None)
+        """
+        cfunc = jit(nopython=True)(numpy_noncentral_chisquare_default)
+        inputs = (
+            (0.5, 1), # test branch when df < 1
+            (1, 5),
+            (5, 1),
+            (100000, 1),
+            (1, 10000),
+        )
+        for input in inputs:
```

Comment:
```suggestion
        for df, nonc in inputs:
```

You can unpack directly in the loop and thus delete the line below.